crates/sui-types/src/execution.rs

```diff
@@ -169,6 +171,23 @@ impl ExecutionResultsV2 {
                     );
                     *initial_shared_version = lamport_version;
                 }
+
+                // Update initial_shared_version for reshared objects
+                if reshare_at_initial_version {
+                    if let Some(Owner::Shared {
+                        initial_shared_version: previous_initial_shared_version,
+                    }) = input_objects.get(id).map(|obj| &obj.owner)
+                    {
+                        debug_assert!(!self.created_object_ids.contains(id));
+                        debug_assert!(!self.deleted_object_ids.contains(id));
+                        debug_assert!(
+                            *initial_shared_version == SequenceNumber::new()
+                                || *initial_shared_version == *previous_initial_shared_version
```

Comment:
why isn't it always == previous_initial_shared_version? i would think the only way it can be SequenceNumber::new() is if that is also what the previous version is?

Reply:
That's the bug here, that sometimes it is `SequenceNumber::new()`

Reply:
This is the issue today  -- in the `written_objects` the `initial_shared_version` will be `SequenceNumber::new()` in this case, and if it's already equal to the initial_shared_version on input there's no issue.